src/sentry/api/endpoints/relocation.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import re
+
+from django.db import router
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, region_silo_endpoint
+from sentry.api.permissions import SuperuserPermission
+from sentry.models.files.file import File
+from sentry.models.relocation import Relocation, RelocationFile
+from sentry.models.user import MAX_USERNAME_LENGTH
+from sentry.services.hybrid_cloud.user.service import user_service
+from sentry.utils.db import atomic_transaction
+
+# Relocation input files are uploaded as tarballs, and chunked and stored using the normal
+# `File`/`AbstractFile` mechanism, which has a hard limit of 2GiB, because we need to represent the
+# offset into it as a 32-bit int. This means that the largest tarball we are able to import at this
+# time is 2GiB. When validating this tarball, we will need to make a "composite object" from the
+# uploaded blobs in Google Cloud Storage, which has a limit of 32 components. Thus, we get our blob
+# size of the maximum overall file size (2GiB) divided by the maximum number of blobs (32): 65536MiB
+# per blob.
```

Comment:
We have `limit-post: 209715200` configured for [uwsgi](https://uwsgi-docs.readthedocs.io/en/latest/Options.html#limit-post) which limits our max upload size. We shouldn't see overly large files in production because of this.

Reply:
That makes sense. I've left this is as is for now, since that setting is far away from this one and could possible change, whereas the 32-bit offset is more integral to the system and seems pretty stable (we'd probably need to do a decent size refactor bump it). 65Mb chunks for these rarely accessed files seems okay to me? I've added a note about the uwsgi limit though.